File: src/main/java/br/com/correios/enderecador/telas/DestinatarioTableModel.java
package br.com.correios.enderecador.telas;

import br.com.correios.enderecador.bean.DestinatarioBean;

import java.util.ArrayList;
import javax.swing.table.DefaultTableModel;

public class DestinatarioTableModel extends DefaultTableModel {
    private final String[] colunas = new String[]{"Empresa/Nome", "Endereço", "Numero/Lote", "Cidade", "CEP", "UF"};

    private ArrayList<DestinatarioBean> destinatarioBean = new ArrayList<>();

    public int getColumnCount() {
        return this.colunas.length;
    }

    public String getColumnName(int i) {
        return this.colunas[i];
    }

    public int getRowCount() {
        int retorno = 0;
        if (this.destinatarioBean != null)
            retorno = this.destinatarioBean.size();
        return retorno;
    }

    public Object getValueAt(int linha, int coluna) {
        DestinatarioBean destinatarioBean = this.destinatarioBean.get(linha);
        String valor = "";
        if (coluna == 0) {
            valor = destinatarioBean.getNome() + " " + destinatarioBean.getApelido();
        } else if (coluna == 1) {
            valor = destinatarioBean.getEndereco() + " " + destinatarioBean.getComplemento() + " " + destinatarioBean.getBairro();
        } else if (coluna == 2) {
            valor = destinatarioBean.getNumeroEndereco();
        } else if (coluna == 3) {
            valor = destinatarioBean.getCidade();
        } else if (coluna == 4) {
            valor = destinatarioBean.getCep();
        } else if (coluna == 5) {
            valor = destinatarioBean.getUf();
        } else {
            valor = "";
        }
        if (valor == null)
            valor = "";
        return valor;
    }

    public boolean isCellEditable(int linha, int coluna) {
        return false;
    }

    public void setDestinatario(ArrayList destinatarioBean) {
        this.destinatarioBean = destinatarioBean;
        if (this.destinatarioBean == null)
            this.destinatarioBean = new ArrayList<>();
        fireTableDataChanged();
    }

    public DestinatarioBean getDestinatario(int linha) {
        DestinatarioBean destinatario = null;
        if (this.destinatarioBean.get(linha) != null)
            destinatario = this.destinatarioBean.get(linha);
        return destinatario;
    }

    public void setDestinatario(int linha, DestinatarioBean destinatarioBean) {
        this.destinatarioBean.set(linha, destinatarioBean);
        fireTableDataChanged();
    }

    public void addDestinatario(DestinatarioBean destinatario) {
        this.destinatarioBean.add(destinatario);
        fireTableDataChanged();
    }

    public void removeDestinatario(DestinatarioBean destinatario) {
        this.destinatarioBean.remove(destinatario);
        fireTableDataChanged();
    }

    public int indexOf(DestinatarioBean destinatario) {
        return this.destinatarioBean.indexOf(destinatario);
    }
}

File: src/main/kotlin/br/com/correios/enderecador/telas/DestinatarioTableModel.kt
package br.com.correios.enderecador.telas

import javax.swing.table.DefaultTableModel
import br.com.correios.enderecador.bean.DestinatarioBean
import java.util.ArrayList

class DestinatarioTableModel : DefaultTableModel() {
    private val colunas = arrayOf("Empresa/Nome", "Endereço", "Numero/Lote", "Cidade", "CEP", "UF")
    private var destinatarioBean: ArrayList<DestinatarioBean?>? = ArrayList()

    override fun getColumnCount(): Int {
        return colunas.size
    }

    override fun getColumnName(i: Int): String {
        return colunas[i]
    }

    override fun getRowCount(): Int {
        var retorno = 0
        if (destinatarioBean != null) retorno = destinatarioBean!!.size
        return retorno
    }

    override fun getValueAt(linha: Int, coluna: Int): String {
        val destinatarioBean = destinatarioBean!![linha]
        return when (coluna) {
            0 -> destinatarioBean!!.nome + " " + destinatarioBean.apelido
            1 -> destinatarioBean!!.endereco + " " + destinatarioBean.complemento + " " + destinatarioBean.bairro
            2 -> destinatarioBean!!.numeroEndereco
            3 -> destinatarioBean!!.cidade
            4 -> destinatarioBean!!.cep.orEmpty()
            5 -> destinatarioBean!!.uf
            else -> ""
        }
    }

    override fun isCellEditable(linha: Int, coluna: Int): Boolean {
        return false
    }

    fun setDestinatario(destinatarioBean: ArrayList<*>?) {
        this.destinatarioBean = destinatarioBean as ArrayList<DestinatarioBean?>?
        if (this.destinatarioBean == null) this.destinatarioBean = ArrayList()
        fireTableDataChanged()
    }

    fun getDestinatario(linha: Int): DestinatarioBean? {
        var destinatario: DestinatarioBean? = null
        if (destinatarioBean!![linha] != null) destinatario = destinatarioBean!![linha]
        return destinatario
    }

    fun setDestinatario(linha: Int, destinatarioBean: DestinatarioBean?) {
        this.destinatarioBean!![linha] = destinatarioBean
        fireTableDataChanged()
    }

    fun addDestinatario(destinatario: DestinatarioBean?) {
        destinatarioBean!!.add(destinatario)
        fireTableDataChanged()
    }

    fun removeDestinatario(destinatario: DestinatarioBean?) {
        destinatarioBean!!.remove(destinatario)
        fireTableDataChanged()
    }

    fun indexOf(destinatario: DestinatarioBean?): Int {
        return destinatarioBean!!.indexOf(destinatario)
    }
}